webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -414,58 +424,51 @@ class InternalLDAPQueryProcessor(
+        case None      => Seq[LDAPEntry]().succeed
+        case Some(dms) =>
+          (for {
+            // Ok, do the computation here
+            _      <- logPure.ifTraceEnabled {
+                        ZIO.foreach_(dms) { case (dnType, dns) =>
+                          logPure.trace(s"/// ${dnType} ==> ${dns.map(_.getRDN).mkString(", ")}")
+                        }
                       }
-                    }
-          fss     = filterSeqSet(dms)
-          blf     <- buildLastFilter(nq, fss)
-
-          // for convenience
-          (finalLdapFilter, finalSpecialFilters) = blf
-
-          //final query, add "match only server id" filter if needed
-          rt      = nodeObjectTypes.copy(filter = finalLdapFilter)
-          _       <- logPure.debug(s"[${debugId}] |- (final query) ${rt}")
-          entries <- (for {
-            con      <- ldap
-            results  <- executeQuery(rt.baseDn, rt.scope, nodeObjectTypes.objectFilter, rt.filter, finalSpecialFilters, select.toSet, nq.composition, debugId)
-            // if we need to invert result, do it here
-            finalRes <- query.transform match {
-                          case ResultTransformation.Identity => results.succeed
-                          case ResultTransformation.Invert   =>
-                            for {
-                              _      <- logPure.debug(s"[${debugId}] |- (need to get all nodeIds for inversion) ")
-                              allIds <- executeQuery(rt.baseDn, rt.scope, nodeObjectTypes.objectFilter, Some(ALL), Set(), Set(), nq.composition, debugId)
-                              ids    =  results.flatMap(x => x(A_NODE_UUID)).toSet
-                              res    =  allIds.collect { case e if(!ids.contains(e.value_!(A_NODE_UUID))) => e }
-                              _      <- logPure.debug(s"[${debugId}] |- (invert) entries after inversion: ${res.size}")
-                            } yield {
-                              res
-                            }
-                          }
-          } yield {
-            postFilterNode(finalRes.groupBy(_.dn).map(_._2.head).toSeq, query.returnType, limitToNodeIds)
-          }).foldM(
-            err =>
-              logPure.debug(s"[${debugId}] `-> error: ${err.fullMsg}") *>
-                err.fail
-            , seq =>
-              logPure.debug(s"[${debugId}] `-> ${seq.size} results") *>
-                seq.succeed
-          )
-        } yield {
-          LdapQueryProcessorResult(entries, nq.nodeInfoFilters)
-        }
+            fss     = filterSeqSet(dms)
+            blf     <- buildLastFilter(nq, fss)
+
+            // for convenience
+            (finalLdapFilter, finalSpecialFilters) = blf
+
+            //final query, add "match only server id" filter if needed
+            rt      = nodeObjectTypes.copy(filter = finalLdapFilter)
+            _       <- logPure.debug(s"[${debugId}] |- (final query) ${rt}")
+            entries <- for {
```

Comment:
We don't need the for here if i read correctly

Reply:
right, it was simplified enought :)